haproxy_stats.sh

```diff
@@ -27,46 +18,11 @@ debug() {
 debug "SOCAT_BIN        => $SOCAT_BIN"
 debug "CACHE_FILEPATH   => $CACHE_FILEPATH"
 debug "CACHE_EXPIRATION => $CACHE_EXPIRATION minutes"
-debug "HAPROXY_SOCKET   => $HAPROXY_SOCKET"
+debug "HAPROXY_SOCK   => $HAPROXY_SOCK"
 debug "pxname   => $pxname"
 debug "svname   => $svname"
 debug "stat     => $stat"
 
-# check if socat is available in path
-if [ -z "$SOCAT_BIN" ] && [ "$GET_STATS" -eq 1 ]
-then
-  echo 'ERROR: cannot find socat binary'
-  exit 126
-fi
-
-# if we are getting stats:
-#   check if we can write to stats cache file, if it exists
-#     or cache file path, if it does not exist
-#   check if HAPROXY socket is writable
-# if we are NOT getting stats:
-#   check if we can read the stats cache file
-if [ "$GET_STATS" -eq 1 ]
-then
-  if [ -e "$CACHE_FILEPATH" ] && [ ! -w "$CACHE_FILEPATH" ]
-  then
-    echo 'ERROR: stats cache file exists, but is not writable'
-    exit 126
-  elif [ ! -w ${CACHE_FILEPATH%/*} ]
-  then
-    echo 'ERROR: stats cache file path is not writable'
-    exit 126
-  fi
-  if [ ! -w $HAPROXY_SOCKET ]
-  then
-    echo "ERROR: haproxy socker is not writable"
-    exit 126
-  fi
-elif [ ! -r "$CACHE_FILEPATH" ]
-then
-  echo 'ERROR: cannot read stats cache file'
-  exit 126
-fi
-
```

Comment:
> and do check_socat in whichever script needs it..

I think it will be useful only with daemonizing checker. But as far as it small script with small execution time - it's possible to use every check once and it will be enough.